File: src/main/java/eladkay/quaritum/common/entity/EntityCircleOfTheFinalMoment.java
package eladkay.quaritum.common.entity;

import net.minecraft.entity.Entity;
import net.minecraft.nbt.NBTTagCompound;
import net.minecraft.util.math.BlockPos;
import net.minecraft.world.World;

public class EntityCircleOfTheFinalMoment extends Entity {

    private static final int RANGE = 16;
    public int ticksLeft;

    public EntityCircleOfTheFinalMoment(World worldIn) {
        super(worldIn);
        ticksLeft = 1000;
    }

    public EntityCircleOfTheFinalMoment(World worldIn, BlockPos pos) {
        super(worldIn);
    }

    @Override
    public void onUpdate() {
        super.onUpdate();
        System.out.println("ticks: " + ticksLeft);
        ticksLeft--;
        if (ticksLeft <= 0) {
            kill();
            worldObj.removeEntity(this);
        }
    }

    @Override
    protected void entityInit() {
        ticksLeft = 1000;
    }

    @Override
    protected void readEntityFromNBT(NBTTagCompound compound) {
        ticksLeft = compound.getInteger("tix");
    }

    @Override
    protected void writeEntityToNBT(NBTTagCompound compound) {
        compound.setInteger("tix", ticksLeft);
    }
}

File: src/main/java/eladkay/quaeritum/common/entity/EntityCircleOfTheFinalMoment.kt
package eladkay.quaeritum.common.entity

import net.minecraft.entity.Entity
import net.minecraft.nbt.NBTTagCompound
import net.minecraft.util.math.BlockPos
import net.minecraft.world.World

class EntityCircleOfTheFinalMoment : Entity {
    var ticksLeft: Int = 0

    constructor(worldIn: World) : super(worldIn) {
        ticksLeft = 1000
    }

    constructor(worldIn: World, pos: BlockPos) : super(worldIn)

    override fun onUpdate() {
        super.onUpdate()
        println("ticks: " + ticksLeft)
        ticksLeft--
        if (ticksLeft <= 0) {
            setDead()
            world.removeEntity(this)
        }
    }

    override fun entityInit() {
        ticksLeft = 1000
    }

    override fun readEntityFromNBT(compound: NBTTagCompound) {
        ticksLeft = compound.getInteger("tix")
    }

    override fun writeEntityToNBT(compound: NBTTagCompound) {
        compound.setInteger("tix", ticksLeft)
    }

    companion object {

        private val RANGE = 16
    }
}
